pandas/core/strings.py

```diff
@@ -35,19 +35,32 @@
 _shared_docs = dict()
 
 
-def _get_array_list(arr, others):
+def _get_array_list(arr, others, align=True):
     from pandas.core.series import Series
 
-    if len(others) and isinstance(com._values_from_object(others)[0],
-                                  (list, np.ndarray, Series)):
+    if (len(arr) and isinstance(arr, Series)
+            and len(others) and isinstance(others, Series)):
+        if align is None:
+            align = False
+            warnings.warn("A future version of pandas will perform alignment "
+                          "when others is a series. To disable alignment (the "
```

Comment:
this is a bit unclear, there is no 'previous behavior; the default is to  not ``align``, but it will change in the future

Reply:
The text is the suggestion of @TomAugspurger in the original issue #18657